src/js/component.js

```diff
@@ -160,7 +160,7 @@ class Component {
    * @param {string|string[]} type
    *        An event name or an array of event names.
    *
-   * @param {Function} fn
+   * @param {Function} [fn]
    *        The function to remove.
```

Comment:
```suggestion
   * @param {Function} [fn]
   *        The function to remove. If not specified, all listeners managed by Video.js will be removed.
```
I think the thing to do it here is to mark it as optional as it is, but warn of the consequences when omitting it.